### PR TITLE
Test: Use different database for FiltersInheritanceTests

### PR DESCRIPTION
**Pull request by smitpatel**
Since InheritanceTests modify the data causing flaky errors
Also add InheritanceTests for Sqlite

@anpete @AndriySvyryd

### CODE REVIEW

**Comment by smitpatel** [comment]
@AndriySvyryd - Is there an example of it?

**Comment by AndriySvyryd** [comment]
@smitpatel Yes, Find usages

**Comment by smitpatel** [comment]
@AndriySvyryd - Updated.

**Comment by smitpatel** [comment]
@AndriySvyryd @anpete - Can you review again? I had to change a lot in test fixture & test base to make tests work with transaction.

File: test/EFCore.SqlServer.FunctionalTests/Query/InheritanceSqlServerFixture.cs [hunk 1]
```diff
@@ -9,34 +10,54 @@ namespace Microsoft.EntityFrameworkCore.Query
...
+                .UseInternalServiceProvider(serviceProvider)
+                .Options;
+        }
 
         public override SqlServerTestStore CreateTestStore()
-            => SqlServerTestStore.GetOrCreateShared(DatabaseName, () =>
-                {
-                    using (var context = CreateContext())
+        {
+            return SqlServerTestStore.GetOrCreateShared(
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/Query/InheritanceSqlServerFixture.cs]: Why not expression body? :trollface:

**Pull request merged**